cmd/clusterctl/client/repository/client.go

```diff
@@ -173,6 +173,14 @@ func repositoryFactory(providerConfig config.Provider, configVariablesClient con
 		return repo, err
 	}
 
+	if rURL.Scheme == httpsScheme {
```

Comment:
I can also think of later to add Github Enterprise support later on, where the above would not match because of the Host.

May it make sense to also check here for the `gitlabPackagesAPIPrefix`?

Reply:
Also: maybe include it with the above if and return a proper error message if it is not github or gitlab:
```go
if rURL.Scheme == httpsScheme {
  // do github
  
  // do gitlab
  
  // return proper error message for https and do not fallback to "invalid provider url..."
}